### PR TITLE
Adding permission schema for query based sync

### PR DESCRIPTION
**Pull request by nhachicha**
## What, How & Why?
This adds the permission schema at the C++ level to cover the various way of opening Realm (`new Realm(config)`, `Realm.open` and `Realm.openAsync()`). 

This closes # https://github.com/realm/realm-js/issues/2016 & https://github.com/realm/realm-js/issues/2017

## ☑️ ToDos

* [x] 📝 Changelog entry
* [x] 📝 `Compatibility` label is updated or copied from previous entry
* [x] 🚦 Tests
* [x] 📝 Public documentation PR created or is not necessary
* [x] 💥 `Breaking` label has been applied or is not necessary

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+2.16.0 Release notes (YYYY-MM-DD)
+=============================================================
+
+### Bug fixes
+* Including user's schema without overriding the permission schema added for query based Realm [#2017](https://github.com/realm/realm-js/issues/2017).
```
Comment (kneth) [inline CHANGELOG.md]: Do you know when the bug was introduced? If you know, please add it as `since v...`.

File: src/js_realm.hpp [hunk 2]
```diff
@@ -501,6 +501,85 @@ void RealmClass<T>::constructor(ContextType ctx, ObjectType this_object, size_t
...
+                    {"role", realm::PropertyType::Object|realm::PropertyType::Nullable, "__Role"}
+                }};
+                objectsSchema.emplace_back(std::move(user));
+            }
+            
+            objectsSchema.insert(objectsSchema.end(), std::make_move_iterator(config.schema->begin()),
+                                                      std::make_move_iterator(config.schema->end()));
+            
+            config.schema.emplace(realm::Schema(objectsSchema));
+            schema_updated = true;
```
Comment (kneth) [inline src/js_realm.hpp]: Also when nothing is added?

Comment (nhachicha) [reply src/js_realm.hpp]: Nice catch!

Comment (nhachicha) [reply src/js_realm.hpp]: fixed

**Comment by bmunkholm** [comment]
@nhachicha Can you try to elaborate the changelog entries with end-user eyes? How could the bug be observed and what the impact?

**Comment by nhachicha** [comment]
@bmunkholm updated

**Pull request merged**